Homework/Khang_Nguyen/sort-colors.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @param {number[]} nums
+ * @return {void} Do not return anything, modify nums in-place instead.
+ */
+ var sortColors = function(nums) {
+    const len = nums.length
+    let i = 0
+    let leftMost1 = -1
+    let leftMost2 = -1
+    
+// nums[i] === 2
+// no leftmost 2 => assign
+
+// nums[i] === 1 
+// - no leftmost 1 => assign then check if have leftmost 1 and 2 => if leftmost 1 > leftmost 2, -> swap leftmost 1 and 2 pos, assign leftmost 2 to leftmost 1, increase leftmost 2 by 1
+// - have leftmost 1 => have leftmost 2 => swap with leftmost 2, increase leftmost 2. 
+
+// nums[i] === 0
+// - have no leftmost => do nothing
+// - have only 1 leftmost => swap, incr leftmost index
+// - have both => swap i with leftmost 1, then swap i with leftmost 2, increase both leftmost by 1.        
+    while (i < len) {
```

Comment:
a simpler solution: move all 0 to left, move all 2 to right.

```
class Solution {
public:
    void sortColors(vector<int>& a) {
        int index = -1;
        for(int i = 0; i < a.size(); i++)
            if (a[i] == 0) {
                index++;
                swap(a[i], a[index]);
            }
        
        for(int i = index+1 ; i < a.size(); i++)
            if (a[i] == 1) {
                index++;
                swap(a[i], a[index]);
            }
    }
};
```

Reply:
Yeah I try to do it with 1 loop only 👯